### DISCUSSION TITLE
How can I access custom parameters of the Route class?

### DISCUSSION THREAD

**Discussion by T3rm1** [molefrog/wouter#437] [Q&A] [answered]
Let's say I have this route: `<Route path="/2023" component={TopPlayers} year="2023" />`
How can I access `year` in `TopPlayers`?

And yes, I don't want it to be dynamic. I know about dynamic patterns.

**Comment by molefrog** [comment, marked as answer]
Like so: 
```jsx
<Route path="/2023">
  <TopPlayers year="2023" />
</Route>
```

**Reply by T3rm1** [reply]
It does work. However, if I also have a dynamic pattern (e.g. `<Route path="/2023/:id">` then I can't access it through `props.params.id`. `props.year` is available. And if I use the `component={}` approach, then it's the other way round. A little inconsistent. Is it really required to do
```
<Route path="/2023/:id" />
  {params => <TopPlayers year="2023" id={params.id} />}
</Route>
```
Maybe this could be fixed in a future release.

**Reply by molefrog** [reply]
In that case, you can use function as children:

```jsx
<Route path="/2023/:id">
  {(params) => <TopPlayers year="2023" params={params} />}
</Route>
